Beta_Integrations/integration-Vertica.yml

```diff
@@ -0,0 +1,175 @@
+description: Analytic database management software
+configuration:
+- display: Host (Vertica Server URL)
+  name: url
+  defaultvalue: myhost.example.com
+  type: 0
+  required: true
+- display: Database
+  name: database
+  defaultvalue: mydb
+  type: 0
+  required: true
+- display: Username
+  name: credentials
+  defaultvalue: ""
+  type: 9
+  required: true
+- display: Port
+  name: port
+  defaultvalue: "5433"
+  type: 0
+  required: false
+script:
+  script: |-
+    ''' IMPORTS '''
+
+    import vertica_python
+    import datetime
+
+    ''' GLOBALS/PARAMS '''
+
+    USERNAME = demisto.params().get('credentials').get('identifier')
+    PASSWORD = demisto.params().get('credentials').get('password')
+    DATABASE = demisto.params().get('database')
+    PORT = int(demisto.params().get('port', 5433))
+    SERVER = demisto.params()['url'][:-1] if (demisto.params()['url'] and demisto.params()['url'].endswith('/')) else demisto.params()['url']
+
+    ''' HELPER FUNCTIONS '''
+
+    def check_if_datetime(v):
+        """
+        Parses datetime object into string
+        """
+        if isinstance(v, datetime.datetime):
+            return v.strftime("%Y-%m-%dT%H:%M:%S")
+        return v
+
+
+    def connect_db():
+        db_params = {
+            'host': SERVER,
+            'port': PORT,
+            'user': USERNAME,
+            'password': PASSWORD,
+            'database': DATABASE,
+            'connection_timeout': 5
+        }
+        try:
+            connection = vertica_python.connect(**db_params)
+        except vertica_python.errors.ConnectionError:
+            return_error('Could not connect to DB, please re-check DB params.')
+        return connection
+
+    CONNECTION = connect_db()
+    CURSOR = CONNECTION.cursor('dict')
+
+    ''' COMMANDS + QUERY FUNCTIONS '''
+
+    def test_module():
+        """
+        Performs basic query on default system tables
+        """
+        CURSOR.execute("SELECT * FROM system_tables ORDER BY table_schema, table_name LIMIT 2;")
+        rows = CURSOR.fetchall()
+        if CURSOR.rowcount == 0:
+            return_error('No results were returned from the DB.')
+
+
+    def query_command():
+        """
+        Execute a query against the DB
+        """
+        # Init main vars
+        contents = []
+        context = {}
+        title = 'Vertica Query'
+        # Get arguments from user
+        query = demisto.args().get('query')
+        limit = int(demisto.args().get('limit', 50))
+        # Query and get raw response (list of ordered dicts)
+        rows = query_request(query)
+        # Parse response into context & content entries
+        if rows:
+            if limit:
+                rows = rows[:limit]
+            title = 'Vertica Query Results:'
```

Comment:
you override the old title?

Reply:
indeed, depends on if we have results or not